### PR TITLE
Feat/check available bumps pelorus operator

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Describe the behavior changes introduced in this PR

Create scheduled work to check if pelorus operator can be bumped.

## Linked Issues?

No linked issues.

## Testing Instructions

To see the new command help, run `make help`
To run new command, run `make check-bump-pelorus-operator`

### CODE REVIEW

File: .github/workflows/check-update-pelorus-operator.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron:  '0 0 * * 1' # Every Monday at 00:00 UTC
```
Comment (mateusoliveira43) [inline .github/workflows/check-update-pelorus-operator.yml:11]: I think weekly is enough, what you think?
I just entered a random date, we can choose when it is best to run it

File: .github/workflows/check-update-pelorus-operator.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Check for updates for pelorus operator
+
+on:
+  push:
```
Comment (mateusoliveira43) [inline .github/workflows/check-update-pelorus-operator.yml:4]: just run on schedule?

File: .github/workflows/check-update-pelorus-operator.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          cache: "pip"
+          cache-dependency-path: |
+            **/requirements*.txt
+            pyproject.toml
+
+      - name: Check if Pelorus operator was bumped
+        run: |
+          make bump-pelorus-operator
+          git restore pelorus-operator/helm-charts/pelorus/Chart.lock
+          test -z "$(git status --short)"
```
Comment (mateusoliveira43) [inline .github/workflows/check-update-pelorus-operator.yml:31]: I will check how to check if the job failed, and if so, inform a message like "Updates can be made to pelorus operator. Run `make bump-pelorus-operator` to update it."

File: .github/workflows/check-update-pelorus-operator.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          python-version: "3.10"
+          cache: "pip"
+          cache-dependency-path: |
+            **/requirements*.txt
+            pyproject.toml
+
+      - name: Check if Pelorus operator was bumped
+        run: |
+          make bump-pelorus-operator
+          git restore pelorus-operator/helm-charts/pelorus/Chart.lock
```
Comment (mateusoliveira43) [inline .github/workflows/check-update-pelorus-operator.yml:30]: here I am ignoring chart.lock changes, but there is still the false positive changes we have to treat (when the script changes yaml sections position)

File: scripts/create_pelorus_operator [hunk 5]
```diff
@@ -115,15 +121,23 @@ pushd "${destination_dir}" || exit 2
     # Correct operator version, to be latest +1
     OPERATOR_VERSION=$(curl -H "Authorization: Bearer XYZ" -X GET "https://quay.io/api/v1/repository/${OPERATOR_ORG_NAME}/${OPERATOR_PROJECT_NAME}/tag/" | jq .tags[].name | head -1 | sed -e 's|\"||g')
     echo "INFO: Current operator version: ${OPERATOR_VERSION}"
-    NEW_OPERATOR_VERSION=$(echo ${OPERATOR_VERSION} | awk -F. -v OFS=. '{$NF += 1 ; print}')
+    if [ ! -z ${file_version} ] && [ ${file_version} != ${OPERATOR_VERSION} ]; then
+        echo "ERROR: file version ${file_version} does not match released version ${OPERATOR_VERSION}."
+        exit 2
+    fi
+    if $no_update; then
```
Comment (mateusoliveira43) [inline scripts/create_pelorus_operator:128]: @mpryc added the logic to not update version, but there are still false positives

File: scripts/create_pelorus_operator [hunk 6]
```diff
@@ -115,15 +121,23 @@ pushd "${destination_dir}" || exit 2
     # Correct operator version, to be latest +1
     OPERATOR_VERSION=$(curl -H "Authorization: Bearer XYZ" -X GET "https://quay.io/api/v1/repository/${OPERATOR_ORG_NAME}/${OPERATOR_PROJECT_NAME}/tag/" | jq .tags[].name | head -1 | sed -e 's|\"||g')
     echo "INFO: Current operator version: ${OPERATOR_VERSION}"
-    NEW_OPERATOR_VERSION=$(echo ${OPERATOR_VERSION} | awk -F. -v OFS=. '{$NF += 1 ; print}')
+    if [ ! -z ${file_version} ] && [ ${file_version} != ${OPERATOR_VERSION} ]; then
+        echo "ERROR: file version ${file_version} does not match released version ${OPERATOR_VERSION}."
```
Comment (mateusoliveira43) [inline scripts/create_pelorus_operator:125]: @mpryc added this to avoid duplicated versions in repo

**Review by KevinMGranger — COMMENTED** [review]
The stuff I commented on looks good. But I'll wait to hear from @mpryc since he's more experienced with the operator stuff.

File: .github/workflows/check-update-pelorus-operator.yml (hunk 2 above)
Comment (mateusoliveira43) [reply .github/workflows/check-update-pelorus-operator.yml:4]: I will change the rule to apply only for schedule and related operator stuff

**Comment by mateusoliveira43** [comment]
discuss after https://github.com/konveyor/pelorus/issues/841 is resolved if this PR still makes sense

**Comment by mateusoliveira43** [comment]
Alright, I think this PR is not necessary

I believe what made me want to create it was that each time you run the create-operator script, you can get a different operator sdk version from [here](https://github.com/operator-framework/operator-sdk/releases). So, we only need to check if there is a newer version of it, instead of creating all this logic.

But I think that our logic is wrong in our scripts today. For example, the latest releases of operator sdk today are 1.26.1 and 1.27.0, and our scripts install 1.26.1 because its newer. We should check which is bigger, right?

My idea is: 
- create a CI job to check if the operator sdk is the newest (and bigger) possible when there are changes in pelorus-operator folder; if not, break and warn the about the newer version available.
- update our installation scripts to install the newest (and bigger) possible versions of each software.

Do you agree? @mpryc @KevinMGranger @weshayutin 

If so, I will close this PR and open issues to address each point.

**Comment by mateusoliveira43** [comment]
@mpryc  but we are not forced to just release bigger versions, are we?

For example, current Python is 3.11, but this does not mean they will not release patches to 3.10, right? I believe this is not a mistake from the software versioning.

**Comment by mpryc** [comment]
> @mpryc but we are not forced to just release bigger versions, are we?
> 
> For example, current Python is 3.11, but this does not mean they will not release patches to 3.10, right? I believe this is not a mistake from the software versioning.

That is correct, however latest tag should in this case point to the 3.11 and not the 3.10-update which may have happened after 3.11 release.

**Comment by mpryc** [comment]
I am closing this PR.

In my opinion it is not required. There was an issue with `operator-sdk` release where the following order was made:


```
$ curl --silent "https://api.github.com/repos/operator-framework/operator-sdk/releases" | jq -r '.[].tag_name'
v1.28.0
v1.26.1
v1.27.0
v1.25.4
v1.26.0
v1.25.3
v1.25.2
v1.25.1
v1.25.0
v1.24.1
v1.24.0
v1.23.0
v1.22.2
v1.22.1
v1.22.0
v1.21.0
v1.20.1
v1.20.0
v1.19.1
v1.19.0
v1.18.1
v1.18.0
v1.17.0
v1.16.0
v1.15.0
v1.14.0
v1.13.1
v1.13.0
v1.12.0
v1.11.0
```

As you can see  there were two times where the latest tag was not the latest number:

```
v1.26.1       <- After  v1.27.0
v1.25.4       <- After  v1.26.0
```

But it's not our responsibility to decide what is the `latest` tag for a particular repository, maybe there was some decision behind making.

Logged but in the operator-sdk project: https://github.com/operator-framework/operator-sdk/issues/6403

Closing this one, we should monitor the answers of the above issue and see if this one will need to be re-opened.

With the latest tag v1.28.0 we are safe, but we need to get answers to not hit this again in the future.

**Pull request closed without merging**